smart-contracts/advanced/deterministic-deployment.md

```diff
@@ -0,0 +1,87 @@
+
+- Using a **factory contract that has been deployed keylessly** to deploy your contracts.
+    - The factory usually uses the `CREATE2` opcode to deploy your contracts, precomputing the deployment address and ensuring consistency across all chains.
+- **Deploying your contracts keylessly** (without using any factory) onto all blockchains.
+    - You must keep the code and constructor arguments unchanged to get the same address.
+
+## Creating a deterministic deployment
+
+Contract address can be precomputed, before the contract is deployed, using a create opcode. This is one of the options for deterministic deployment.
+
+### **`CREATE`**
+
+If deploying using a `CREATE` factory contract, contract addresses are calculated using the:
+
+- Address of the factory contract (that calls `CREATE` opcode) itself;
+- Nonce of the factory contract ([EIP-161](https://eips.ethereum.org/EIPS/eip-161) specifies that contract nonce starts at 1, not 0 like EOAs).
+
+### **`CREATE2`**
+
+Similar to `CREATE`, `CREATE2` is used to deploy contracts to the same address across different blockchain networks in factory contract implementations. Contract addresses are precomputed using the:
+
+- Address of the factory contract (that calls `CREATE2` opcode) itself;
+- Bytecode of the contract;
+- Salt (a chosen value).
+
+See this [code snippet](https://solidity-by-example.org/app/create2/) for an example of how to call `CREATE2` from a factory contract.
+
+## **Deploying a factory contract to Filecoin**
+
+Other people may have already deployed the factory contract onto Filecoin, in which case you won't need to redeploy it. You can just use the factory to deploy your contracts. So first check the expected address on a blockchain explorer to see if a factory contract already exists there.
+
+If there isn't one yet then you'll need to deploy the factory contract via a **reusable signed raw deployment transaction**. The factory contract will then have the same address as on other blockchains (as long as the transaction bytecode stays the same). See the steps below to deploy the factory.
+
+1. **Prepare the deployment transaction:**
+    - Write the smart contract code for the factory contract.
+    - Compile the contract to get the bytecode.
+    - Create a deployment transaction with the bytecode.
+
+2. **Sign the deployment transaction:**
+    - Use a private key to sign the deployment transaction.
+    - Ensure the private key is securely stored and not exposed.
+
+3. **Broadcast the signed transaction:**
+    - Send the signed transaction to the desired blockchain network.
+    - Wait for the transaction to be mined and confirmed.
+
+4. **Verify the deployment:**
+    - Check the [blockchain explorer](https://docs.filecoin.io/networks/mainnet/explorers) to verify that the factory contract has been deployed to the expected address.
+    - Ensure the contract code matches the expected bytecode.
+
```

Comment:
Why not make use of one of the factory contracts listed at the bottom of the document and make this an actual deployment example/tutorial, instead of just a list of steps. I don't think there is much benefit in stating `Write the smart contract code for the factory contract` in a document that, as far as I can tell, is aimed at people who don't necessarily know what deterministic deployments are at the start of the document.